saleor/graphql/menu/types.py

```diff
@@ -0,0 +1,23 @@
+from graphene import relay
+
+from ...menu import models
+from ..core.types import CountableDjangoObjectType
+
+
+class Menu(CountableDjangoObjectType):
+    class Meta:
+        description = """Represents a single menu - an object that is used
+        to help navigate through the store."""
+        interfaces = [relay.Node]
+        filter_fields = ['id', 'name']
+        model = models.Menu
+
+
+class MenuItem(CountableDjangoObjectType):
```

Comment:
One problem with this type is that the consumer of the API has to fetch all of the related objects (page, collection, category) in order to get the actual link. Instead, the type should look more or less like this:

```
type MenuItem {
  id: ID!
  title: String!
  url: String!
}
```

where `url` is automatically derived from the model by the backend. The client shouldn't have to worry about which relation is actually when he only needs the proper URL.